examples/multipage/src/app.rs

```diff
@@ -1,45 +1,29 @@
-use crate::routes;
+use crate::{Route, RouteDispatcher};
 use mogwai::prelude::*;
 
-#[derive(Clone, Copy, Debug, PartialEq)]
-pub enum Route {
-    Home,
-    NotFound,
-}
-
-impl From<Route> for View<HtmlElement> {
-    fn from(route: Route) -> Self {
-        ViewBuilder::from(route).into()
-    }
-}
-
-impl From<Route> for ViewBuilder<HtmlElement> {
-    fn from(route: Route) -> Self {
-        match route {
-            Route::Home => routes::home(),
-            Route::NotFound => routes::not_found(),
-        }
-    }
-}
-
-#[derive(Clone, Copy, Debug, PartialEq)]
+#[derive(Copy, Clone, Debug)]
 pub enum Out {
-    Render(Route),
+    Render { route: Route },
     RenderClicks(i32),
 }
 
 #[derive(Debug)]
 pub struct App {
     click_count: i32,
     current_route: Route,
+    dispatch: RouteDispatcher,
 }
 
 impl App {
-    pub fn new() -> Self {
-        App {
+    pub fn gizmo(initial_route: Route) -> Gizmo<Self> {
+        let tx_model = Transmitter::new();
+        let rx_view = Receiver::new();
+        let app = App {
             click_count: 0,
-            current_route: Route::Home,
-        }
+            current_route: initial_route,
+            dispatch: RouteDispatcher::new(tx_model.clone()),
+        };
+        Gizmo::from_parts(app, tx_model, rx_view)
```

Comment:
This is leftover from https://github.com/schell/mogwai/pull/51/commits/1c4e6afdf255942dd9848c5b5f5b4ea34021470f I looked at `Component::bind` but it would only binds to components that exist in the view initially. And because (as of https://github.com/schell/mogwai/pull/51/commits/1c4e6afdf255942dd9848c5b5f5b4ea34021470f) I was dynamically adding a `Component` into the `View` I didn't think it would continue to work. It is also possible that I simply do not follow how `Component::bind` works and so misunderstood its behavior. However, in this example as it is at PR this is not necessary and I'll update it.

Reply:
I've paired this down so it only creates the Gizmo from the created `App`.